server/group.go

```diff
@@ -58,6 +61,41 @@ func (a *applicationHandler) CreateGroup(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	if util.IsStringEmpty(string(newGroup.Config.Strategy.Type)) {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid strategy type", http.StatusBadRequest))
+		return
+	}
+
+	if string(newGroup.Config.Strategy.Type) != string(config.DefaultStrategyProvider) {
+		_ = render.Render(w, r, newErrorResponse(fmt.Sprintf("unsupported strategy type: %s", newGroup.Config.Strategy.Type), http.StatusBadRequest))
+		return
+	}
+
+	if newGroup.Config.Strategy.Default.IntervalSeconds == 0 {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid interval seconds", http.StatusBadRequest))
+		return
+	}
+
+	if newGroup.Config.Strategy.Default.RetryLimit == 0 {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid retry limit", http.StatusBadRequest))
+		return
+	}
+
+	if util.IsStringEmpty(string(newGroup.Config.Signature.Header)) {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid signature header", http.StatusBadRequest))
+		return
+	}
+
+	if util.IsStringEmpty(string(newGroup.Config.Signature.Hash)) {
+		_ = render.Render(w, r, newErrorResponse("please provide a valid hash", http.StatusBadRequest))
+		return
+	}
```

Comment:
is this necessary since we're still going to check algo.M for it?

Reply:
I think this improves the experience of the API since we'll be returning the correct validation message.

Reply:
Yeah, true.